src/Microsoft.Extensions.Logging.EventSource/EventSourceLoggerProvider.cs

```diff
@@ -0,0 +1,238 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics.Tracing;
+
+namespace Microsoft.Extensions.Logging.EventSourceLogger
+{
+    /// <summary>
+    /// The provider for the <see cref="EventSourceLogger"/>.
+    /// </summary>
+    internal class EventSourceLoggerProvider : ILoggerProvider
+    {
+        // A small integer that uniquely identifies the LoggerFactory assoicated with this LoggingProvider.
+        // Zero is illegal (it means we are uninitialized), and have to be added to the factory. 
+        private int _factoryID;
+
+        private ILoggerFactory _loggerFactory;
+        private LogLevel _defaultLevel;
+        private string _filterSpec;
+        private EventSourceLogger _loggers; // Linked list of loggers that I have created
+
+        public EventSourceLoggerProvider(ILoggerFactory loggerFactory, EventSourceLoggerProvider next = null)
+        {
+            if (loggerFactory == null)
+            {
+                throw new ArgumentNullException(nameof(loggerFactory));
+            }
+            _loggerFactory = loggerFactory;
+            Next = next;
+        }
+
+        public readonly EventSourceLoggerProvider Next;
+
+        /// <summary>
+        /// <inheritdoc />
+        /// </summary>
+        public ILogger CreateLogger(string categoryName)
+        {
+            var newLogger = _loggers = new EventSourceLogger(categoryName, _factoryID, _loggers);
```

Comment:
This code needs to be thread-safe. Multiple requests can be occurring simultaneously and the head of the linked list will be corrupted. in case of a race.